sqlmesh/core/macros.py

```diff
@@ -1300,7 +1300,7 @@ def _coerce(
     """Coerces the given expression to the specified type on a best-effort basis."""
     base_err_msg = f"Failed to coerce expression '{expr}' to type '{typ}'."
     try:
-        if typ is None or typ is t.Any:
+        if typ is None or typ is t.Any or not isinstance(expr, exp.Expression):
```

Comment:
Doesn't this change make the branches that check `str`, `int` etc unreachable?

Reply:
what do you mean? this is checking if expr is an expression, not the base

Reply:
Ah apologies, I misread this. All good.